_data/fellow_projects/automating_awkward_array_testing.yml

```
@@ -0,0 +1,10 @@
+open: yes
+postdate: 2021-04-01
+title: Automating Awkward Array testing
+description: >
+  Awkward Array is a popular library for nested, variable-sized data, including arbitrary-length lists, records, mixed types, and missing data, using NumPy-like idioms. The Awkward Array kernels are being developed for both CPU and GPU. A successful candidate will participate in automating the process of writing and testing the kernel code. Basic knowledge of Python is required.
+Leveraging on the Hypothesis' ghostwriter the project proposes to automate writing the tests developed for both CPU and GPU kernels, measure quality and effectiveness of the tests, develop strategies for automating coverage and property-based testing. Traditional coverage criteria over CPU programs is not adequate over GPU kernels as it uses a completely different programming model and the faults encountered may be specific to the GPU architecture.
+Hypothesis is a Python library for creating unit tests which are simpler to write and more powerful when run, finding edge cases in your code you wouldn’t have thought to look for. It is stable, powerful and easy to add to any existing test suite.
```

```suggestion
  Leveraging on the Hypothesis' ghostwriter the project proposes to automate writing the tests developed for both CPU and GPU kernels, measure quality and effectiveness of the tests, develop strategies for automating coverage and property-based testing. Traditional coverage criteria over CPU programs is not adequate over GPU kernels as it uses a completely different programming model and the faults encountered may be specific to the GPU architecture.
  Hypothesis is a Python library for creating unit tests which are simpler to write and more powerful when run, finding edge cases in your code you wouldn’t have thought to look for. It is stable, powerful and easy to add to any existing test suite.
```

Or rewrap further, but the indent needs to stay to count as part of the description block.